controller.cpp

```diff
@@ -28,29 +27,47 @@ Ctrl* Ctrl::getInstance(){
   return instance;
 }
 
-bool Ctrl::runCalculation(){
+std::vector<std::tuple<std::string, int, double>> Ctrl::loadInputFiles(){
+  // create an instance of the model class
+  // ensures, that there is only ever one instance of the model class
+  if(current_file_loading == NULL){
+    current_file_loading = new Model();
+  }

```

Comment:
Well I guess you found a way to get around the `ensures, that there is only ever one instance of the model class`. The idea way that there is exactly one instance of the model class, that the controller can communicate with, so that the user can manipulate the calculation from different points of entry. For example, a "load" button may already collect some information from the GUI and save them in the attributes of the Model instance. Then when the user presses the "calculate" button, the model can access these attributes.

That's why I don't understand the need for a second Model instance. I've changed it to use the same Model instance as the "calculate" button, and it seems to work just as well. Let me know if there is a reason.

Reply:
I was afraid that two button pressing events calling each a single model could lead to undesirable confusion since I wasn't sure yet of what I would put in the click load file event.
Therefore, I first made the click load file event based on the click start event (including the creation of a new instance of the model class) and made everything separated to avoid bad interactions or confusion.
It is mainly due to my lack of understanding of what actually happens in the initial code you made and how it could be modified in the most efficient manner.